content/blog/getting-started-with-hpe-greenlake-api-for-backup-and-recovery.md

```diff
@@ -0,0 +1,591 @@
+   ```
+
+![API to discover backup for recovery of VM](/img/api-to-discover-vm-for-recovery.png)
+
+2. Let's obtain the Cloud Recovery Point id for the cloud protection recovery from the virtual machine id from the step #1 using the HPE GreenLake [API](https://developer.greenlake.hpe.com/docs/greenlake/services/backup-recovery/public/openapi/backup-recovery-public-v1beta1/operation/VirtualMachineBackupList/) `GET /backup-recovery/v1beta1/virtual-machines/:id/backups` given the virtual machine id `{{vmId}}`. Copy the `{{backupId}}` from the response body from the below figure so that it can be used in the subsequent API execution. The API used for this: 
+
+   ```shellsession
+   GET /backup-recovery/v1beta1/virtual-machines/{{vmId}}/backups?select=name,description,backupType,id
+   ```
+
+![API to obtain the backup Id of a cloud recovery point](/img/api-to-obtain-backup-id-for-recovery.png)
+
+3. Let's obtain the `datastore id` and the `cluster id` from the existing VMFS datastore that can accommodate the datastore type that this VM can be restored. In this hypervisor, I am using the datastore with the name `“0-BRS-VMFS-Test3”` and enter that as part of the filter into HPE GreenLake [API](https://developer.greenlake.hpe.com/docs/greenlake/services/virtualization/public/openapi/virtualization-public-v1beta1/operation/DatastoresList) `GET /virtualization/v1beta1/datastores`. The API used for this: 
+
+   ```shellsession
+   GET /virtualization/v1beta1/datastores?filter=displayName eq '0-BRS-VMFS-Test3'&select=clusterInfo,datastoreType,name,id
+   ```
+
+![API to obtain the cluster and datastore Ids](/img/api-obtain-cluster-and-datastore.png)
+
+4. To obtain the hypervisor Network Id that is required to recover the recovery point into a new virtual machine, I had to discover the hypervisor id. The [API](https://developer.greenlake.hpe.com/docs/greenlake/services/virtualization/public/openapi/virtualization-public-v1beta1/operation/HypervisorManagerList/) used for this:
+
+   ```shellsession
+   GET /virtualization/v1beta1/hypervisor-manager?select=name,id,state,status,dataOrchestratorInfo,services,hypervisorManagerType,releaseVersion&filter=state eq "OK" and status eq "OK" and name eq "vCenter Name"
+   ```
+
+![API to get hypervisorId](/img/api-obtain-hypervisor-id.png)
+
+5. To set the virtual machine network to the correct network port group, I glanced into the existing virtual machine `"0-Linux-Demo-V02"`, discovered the network port group `“Mgmt-DPortGroup”`. From the list of the network port group, I selected the associated port group of the virtual machine by filtering the name of the port group. Afterward, we copied the `<hypervisor-network-id>` using this from the response body of this [API](https://developer.greenlake.hpe.com/docs/greenlake/services/virtualization/public/openapi/virtualization-public-v1beta1/operation/HypervisorNetwork/) to be used for subsequent execution. The API used for this:  
```

Comment:
For the same reason, you'll need to change "I glanced into" to "you'll need to look into..., to discover the network..."

And so forth... here you have "I selected" again, you'd have to change this.

Consider again whether this whole section should be giving instructions or if you want to tell the story of how you did it and make the appropriate modifications.